src/primitive/polyhedron.rs

```diff
@@ -169,6 +175,49 @@ where
     }
 }
 
+/// Iterate over polyhedron faces.
+/// Yields a tuple with the positions of the 3 vertices of each face
+pub struct FaceIterator<'a, S: 'a>
+where
+    S: BaseFloat,
+{
+    polyhedron: &'a ConvexPolyhedron<S>,
+    current: usize,
+}
+
+impl<'a, S: 'a> FaceIterator<'a, S>
+where
+    S: BaseFloat,
+{
+    pub fn new(polyhedron: &'a ConvexPolyhedron<S>) -> Self {
+        Self {
+            polyhedron,
+            current: 0,
+        }
+    }
+}
+
+impl<'a, S> Iterator for FaceIterator<'a, S>
+where
+    S: BaseFloat,
+{
+    type Item = (&'a Point3<S>, &'a Point3<S>, &'a Point3<S>);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.current >= self.polyhedron.faces.len() {
+            None
+        } else {
+            self.current += 1;
+            let face = &self.polyhedron.faces[self.current - 1];
```

Comment:
would be simpler to get a face first, and then increase the current id

Reply:
Possibly.